disk_objectstore/utils.py

```diff
@@ -615,6 +615,17 @@ def read(self, size: int = -1) -> bytes:
         self._update_pos()
         return stream
 
+    def readline(self, size: int = -1) -> bytes:
+        """Read one line without crossing this object's boundary."""
+        remaining = self._length - self._pos
+        if remaining <= 0:
+            return b''
+
+        readline_size = remaining if (size is None or size < 0) else min(size, remaining)
+        line = self._fhandle.readline(readline_size)
```

Comment:
So this reads only up until some terminator, at least for BinaryIO this is `b'\n'`, even though it can be changed for some file handler, the `readline_size` is just a maximum value up until it reads. I am not sure if this is the intended behavior, since my understanding of the `PackedObjectReader` is that it contains already "one line". It seems like you just make reading slower since you randomly chunk your data whenever the you find a `b'\n'` which has no meaning for arbitrary packed objects (unlike your text example in the tests). What is more meaningful is to have some maximum number to limit memory consumption.

Reply:
Thanks Alex!
I generally agree and was also thinking if `b'\n'` really makes sense, or something "fixed". In the end, I followed the "new-line" approach, based on what I saw for the `readline` method of the `BytesIO` / `BufferedReader` that would be used when the repository isn't packed https://github.com/python/cpython/blob/main/Lib/_pyio.py#L509-L518. Moreover,  think that I was mostly biased by the pickle application of that. I just had a quick look there and it seems that `pickle` is also most relying on `b'\n'` in the readline method. 
But again, I agree that this might not be the most reasonable approach for such a packed file.